app/models/container_label_tag_mapping.rb

```diff
@@ -12,5 +12,73 @@ class ContainerLabelTagMapping < ApplicationRecord
   #   We then also add a specific `label_value`->specific `tag` mapping here.
   belongs_to :tag
 
-  validates :label_name, :presence => true
+  # An empty string value is legal for labels; only NULL represents any value.
+  def any_value?
+    label_value.nil?
+  end
+
+  def specific_value?
+    !any_value?
+  end
+
+  def applies_to_label?(label)
+    ((labeled_resource_type.nil? || label.resource_type == labeled_resource_type) &&
+     label.name == label_name &&
+     (any_value? || label.value == label_value))
+  end
+
+  # Returns a Tag or nil.
+  def tag_for_label(label)
+    specific_mapping_for_value(label.value).tag if applies_to_label?(label)
+  end
+
+  # If this is an open ended any-value mapping, finds or creates a
+  # specific-value mapping to a specific tag.
+  def specific_mapping_for_value(value)
+    if specific_value?
+      self
+    else
+      self.class.where(
+        :labeled_resource_type => labeled_resource_type,
+        :label_name            => label_name,
+        :label_value           => value
+      ).first_or_create do |mapping|
+        mapping.tag = create_tag(value)
+      end
+    end
+  end
+
+  def create_tag(value)
```

Comment:
On DB level you can probably attach a category tag to an entity (I think I saw it done in one place, can't find now), but I expect some things will break, esp. UI.  But why?  If you want a known tag, use a known tag.

@simon3z @bazulay You have a better picture of use cases for auto-tagging, can you comment on following assumptions I had:
- If one wants to tag from labels with given key and any values (eg. zone=*):
  - Mapping to a distinct tag per value (e.g. zone=east => Zone: East, zone=west => Zone: West) is useful.
  - Mapping all values to one tag (e.g. zone=east, zone=west => Zone: Known) is not useful?
    (I can add support within current schema, just didn't think of it before.)